pkgs/build-support/node/fetch-yarn-deps/yarn-install-hook.sh

```diff
@@ -0,0 +1,76 @@
+# shellcheck shell=bash
+
+yarnInstallHook() {
+    echo "Executing yarnInstallHook"
+
+    runHook preInstall
+
+    local -r packageOut="$out/lib/node_modules/$(@jq@ --raw-output '.name' package.json)"
+
+    # yarn pack does not work at all with bundleDependencies.
+    # Since we are imediately unpacking, we can just remove them from package.json
+    # This will NOT be fixed in yarn v1: https://github.com/yarnpkg/yarn/issues/6794
+    mv package.json "$TMPDIR/package.json.orig"
+    # Note: two spellings are accepted, 'bundleDependencies' and 'bundledDependencies'
+    @jq@ 'del(.bundleDependencies)|del(.bundledDependencies)' "$TMPDIR/package.json.orig" > package.json
+
+    # TODO: figure out a way to avoid redundant compress/decompress steps
+    yarn pack --filename "$TMPDIR/package-dist.tgz" \
+        --ignore-engines \
+        --ignore-platform \
+        --ignore-scripts \
+        --non-interactive \
+        --offline
+
+    mkdir -p "$packageOut"
+    tar -xzf "$TMPDIR/package-dist.tgz" \
+        -C "$packageOut" \
+        --strip-components 1 \
+        package/
```

Comment:
Here's an idea: Use stdout as the `--filename` (somehow), and pipe that directly to `tar -xzf`..

Reply:
`--filename` does not accept the common `-`, but passing `/dev/stdout` does work.

This ends up creating a corrupted file because yarn still prints other logs to stdout, so I needed to add the `--silent` flag.

Reply:
> `--filename` does not accept the common `-`, but passing `/dev/stdout` does work.
> 
> This ends up creating a corrupted file because yarn still prints other logs to stdout, so I needed to add the `--silent` flag.

Hmm I worry that it is not that platform independent... We should be sure that at least on Darwin it works. A nixpkgs-review report from Darwin will be great...

Reply:
`element-call` fails in the same way.

Reply:
Reverted to creating a file for now. I'm not sure there are any alternatives.